src/main/java/com/cburch/logisim/analyze/model/Implicant.java

```diff
@@ -57,190 +61,260 @@ public void remove() {
     }
   }
 
-  static List<Implicant> computeMinimal(int format, AnalyzerModel model, String variable) {
-    final var table = model.getTruthTable();
-    final var column = model.getOutputs().bits.indexOf(variable);
-    if (column < 0) return Collections.emptyList();
-
-    final var desired = format == AnalyzerModel.FORMAT_SUM_OF_PRODUCTS ? Entry.ONE : Entry.ZERO;
-    final var undesired = desired == Entry.ONE ? Entry.ZERO : Entry.ONE;
-
-    // determine the first-cut implicants, as well as the rows
-    // that we need to cover.
-    final var base = new HashMap<Implicant, Entry>();
-    final var toCover = new HashSet<Implicant>();
-    var knownFound = false;
-    for (var i = 0; i < table.getRowCount(); i++) {
-      final var entry = table.getOutputEntry(i, column);
-      if (entry == undesired) {
-        knownFound = true;
-      } else if (entry == desired) {
-        knownFound = true;
-        final var imp = new Implicant(0, i);
-        base.put(imp, entry);
-        toCover.add(imp);
-      } else {
-        base.put(new Implicant(0, i), entry);
-      }
+  private static int getNrOfOnes(int value, int nrOfBits) {
+    var nrOfOnes = 0;
+    var mask = 1;
+    for (var bitIndex = 0; bitIndex < nrOfBits; bitIndex++) {
+      if ((value & mask) != 0)
```

Comment:
This was marked as resolved but I am not seeing the changes. Can we have line 64 either as one-liner

```java
if ((value & mask) != 0) nrOfOnes++;
```

or with block brackets to avoid confusion

```java
if ((value & mask) != 0) {
    nrOfOnes++;
}
```

Reply:
Sorry my mistake! I missed this one, fixed.